src/elements/content-sidebar/activity-feed/app-activity/AppActivity.js

```diff
@@ -68,54 +65,107 @@ function mapActivityNodes(node: HTMLLinkElement): React.Node {
+            currentUser,
+            error,
+            intl,
+            isPending,
+            permissions,
+        } = this.props;
+
+        const canDelete = getProp(permissions, 'can_delete', false) || currentUser.id === createdBy.id;
+        const createdAtTimestamp = new Date(createdAt).getTime();
+        const isMenuVisible = canDelete && !isPending;
+        const { isConfirmingDelete } = this.state;
+
+        return (
+            <Media
+                className={classNames('bcs-AppActivity', {
                     'bcs-is-pending': isPending || error,
                 })}
+                data-resin-target="loaded"
+                data-resin-feature={`app_activity_${templateId}_card`}
             >
-                <img
-                    className="bcs-app-activity-icon"
-                    alt={intl.formatMessage(messages.appActivityAltIcon, { appActivityName: name })}
-                    src={icon_url}
-                />
-                <div className="bcs-app-activity-content">
-                    <div className="bcs-app-activity-headline">
-                        <span className="bcs-app-activity-app-name">{name}</span>
-                        {canDelete && !isPending && (
-                            <InlineDelete
-                                id={id}
-                                permissions={permissions}
-                                message={<FormattedMessage {...messages.appActivityDeletePrompt} />}
-                                onDelete={onDelete}
-                            />
-                        )}
-                    </div>
+                <Media.Figure>
+                    <img
+                        className="bcs-AppActivity-icon"
+                        alt={intl.formatMessage(messages.appActivityAltIcon, { appActivityName: name })}
+                        src={icon_url}
+                    />
+                </Media.Figure>
+
+                <Media.Body className="bcs-AppActivity-body">
+                    {isMenuVisible && (
+                        <TetherComponent
+                            attachment="top right"
+                            className="bcs-AppActivity-confirm"
+                            constraints={[{ to: 'scrollParent', attachment: 'together' }]}
+                            targetAttachment="bottom right"
+                        >
+                            <Media.Menu isDisabled={isConfirmingDelete}>
+                                {canDelete && (
+                                    <MenuItem onClick={this.handleDeleteClick}>
+                                        <IconTrash color={bdlGray80} />
+                                        <FormattedMessage {...deleteMessages.deleteLabel} />
+                                    </MenuItem>
+                                )}
+                            </Media.Menu>
+
+                            {isConfirmingDelete && (
+                                <CommentDeleteConfirmation
```

Comment:
Should we rename this for the future? `CommentDeleteConfirmation` seems misleading

Reply:
I think that can be deferred to a future refactor where we can put some more effort into sharing behavior between activity types.